Proiect HTML CSS

ChrisTheSpud opened the pull request.

ChrisTheSpud committed:
First Commit

ChrisTheSpud committed:
Revert "First Commit"

This reverts commit 28bfc12b675314333420723454492f76a5d84aae.

ChrisTheSpud committed:
Revert "Revert "First Commit""

This reverts commit bdecfad0b9ce80064b77c316d97a8c504e06c262.

ChrisTheSpud committed:
Proiect HTML CSS

ChrisTheSpud committed:
Update Fonts

Am adaugat un fallback font.

vladutcornel commented on Scripts/backtotop.js:
```diff
@@ -0,0 +1,17 @@
+} else {
+button.style.display = "none";
+}
+}
```
Formatează codul

vladutcornel commented on Styles/styles-beach.css:
```diff
@@ -0,0 +1,128 @@
+    margin: auto;
+}
+
+:is(.image1, .image2) {
```
având în vedere că nu mai e nimic pe lângă, poate ar merge și fără `:is`.

```suggestion
.image1, .image2 {
```

vladutcornel commented on Styles/styles-beach.css:
```diff
@@ -0,0 +1,128 @@
+    grid-row: 1/-1;
+}
+
+:is(.stacked1, .stacked2, .stacked3) > :is(.image3, .image4, .image5) {
```
Aici are mai mult sens `:is`

vladutcornel commented on Styles/styles-beach.css:
```diff
@@ -0,0 +1,128 @@
+    border-radius: 1.25rem;
+    object-fit: cover;
+    margin-inline-start: auto;
+    margin-inline-end: auto;
```
Poți simplifica aici, fiind aceeași valoare.

```suggestion
    margin-inline: auto;
```

vladutcornel replied:
Chiar dacă am atras atenția aici, problema apare și în alte părți.

vladutcornel commented on Styles/styles-beach.css:
```diff
@@ -0,0 +1,128 @@
+    margin-inline-end: auto;
+}
+
+:is(.stacked1, .stacked2, .stacked3) {
```
Din ce văd eu, cele 3 clase fac același lucru. N-ai putea folosi o singură clasă?

Cu ocazia asta, scapi și de numere.

vladutcornel commented on beach.html:
```diff
@@ -0,0 +1,103 @@
+        </header>
+        <main>
+            <div class="container">
+                <section>
+                    <div class="rows">
+                        <div>
+                            <div>
+                                <h2 class="headings">About</h2>
+                            </div>
+                            <div class="paragraph-style">
```
Aici cred că folosești cam multe elemente în HTML. Ai putea încerca ceva cu grid.

vladutcornel commented on index.html:
```diff
@@ -0,0 +1,130 @@
+        <script src="Scripts/script.js"></script>
+    </head>
+    <body>
+        <button onclick="topFunction()" id="topButton" title="Return to top">
```
Nu se mai folosește atributul `onclick` în HTML. În schimb, ar trebui să faci `addEventListener`

vladutcornel commented on index.html:
```diff
@@ -0,0 +1,130 @@
+                <path d="M17.5 40V9.5625L3.5 23.5625L0 20L20 0L40 20L36.5 23.5625L22.5 9.5625V40H17.5Z" fill="white" fill-opacity="0.4"/>
+            </svg>
+        </button>
+        <script src="Scripts/backtotop.js"></script>
```
Scripturile se pun ori în `<head>`, ori chiar înainte de `</body>`.

vladutcornel commented on Scripts/backtotop.js:
```diff
@@ -0,0 +1,17 @@
+let button = document.getElementById("topButton");
+
+window.onscroll = function() {scrollFunction()};
```
Scrie event-urile așa cum v-am arătat

```suggestion
window.addEventListener('scroll', function() {scrollFunction()});
```

Chestiile asta cu `.on...` sunt un sistem mai vechi de a înregistra un event.

vladutcornel commented on Styles/styles-index.css:
```diff
@@ -0,0 +1,129 @@
+}
+
+.paragraph-style {
+    margin: auto;
```
Vezi dacă unii selectori ar putea fi puși în `styles-global`.

Chiar dacă nu reprezintă exact același element pe fiecare pagină, dacă sunt afișate identic pot fi reprezentate de o singură clasă.  
Și chiar dacă nu sunt identice, poate sunt suficient de asemănătoare. Atunci ai pune clasa și în global (proprietățile comune) și CSS-ul dedicat unei pagini ai pune modificările.

ChrisTheSpud committed:
Update CSS + JS

Am incercat sa modific ceea ce a fost mentionat in review, cu cateva exceptii.

ChrisTheSpud committed:
Update beaches.css and other

Am schimbat la pagina beach.html ca pe desktop sa fie sub forma de grid, eliminand astfel o tura de tag-uri <div></div>, iar in restul fisierelor CSS, ceea ce am mai observat eu cu inline si block care ar merge pe o singura linie, am modificat.
